api/src/main/java/com/datastrato/gravitino/tag/TagOperations.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.tag;
+
+import com.datastrato.gravitino.annotation.Evolving;
+import com.datastrato.gravitino.exceptions.NoSuchTagException;
+import com.datastrato.gravitino.exceptions.TagAlreadyExistsException;
+import java.util.Map;
+
+/**
+ * Interface for supporting global tag operations. This interface will provide tag listing, getting,
+ * creating, and other tag operations under a metalake. This interface will be mixed with
+ * GravitinoMetalake or GravitinoClient to provide tag operations.
+ */
+@Evolving
+public interface TagOperations {
+
+  /**
+   * List all the tag names under a metalake.
+   *
+   * @return The list of tag names.
+   */
+  String[] listTags();
+
+  /**
+   * List all the tags with detailed information under a metalake.
+   *
+   * @param extended If true, the extended information of the tag will be included.
+   * @return The list of tags.
+   */
+  Tag[] listTagsInfo(boolean extended);
+
+  /**
+   * Get a tag by its name under a metalake.
+   *
+   * @param name The name of the tag.
+   * @return The tag.
+   * @throws NoSuchTagException If the tag does not exist.
+   */
+  Tag getTag(String name) throws NoSuchTagException;
```

Comment:
Do we need `extended`?

Reply:
Currently, there's no such requirement based on our UI requirement, I think we can add it later on.